Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -967,4 +980,455 @@ open class NavigationMapView: UIView {
+                if let cachedEntry = cachedAnnotationFeature(for: labelText) {
+                    featurePoint = cachedEntry.feature
+                } else {
+                    featurePoint = Feature(Point(maneuverLocation))
+
+                    let tailPosition = AnnotationTailPosition.center
+
+                    // set the feature attributes which will be used in styling the symbol style layer
+                    featurePoint.properties = ["highlighted": true, "tailPosition": tailPosition.rawValue, "text": labelText, "imageName": "AnnotationCentered-Highlighted", "sortOrder": 0]
+
+                    annotationCache?.setValue(feature: featurePoint, coordinate: maneuverLocation, intersection: nil, for: labelText)
+                }
+                features.append(featurePoint)
+            }
+        }
+
+        guard let intersectionsToAnnotate = intersectionsToAnnotate else { return }
+        for (index, intersection) in intersectionsToAnnotate.enumerated() {
+            guard let coordinate = intersection.annotationPoint else { continue }
+            var featurePoint: Feature
+
+            if let intersectingWayName = intersection.intersectingWayName, let cachedEntry = cachedAnnotationFeature(for: intersectingWayName) {
+                featurePoint = cachedEntry.feature
+            } else {
+                featurePoint = Feature(Point(coordinate))
+
+                let tailPosition = intersection.incidentAngle < 180 ? AnnotationTailPosition.left : AnnotationTailPosition.right
+
+                let imageName = tailPosition == .left ? "AnnotationLeftHanded" : "AnnotationRightHanded"
+
+                // set the feature attributes which will be used in styling the symbol style layer
+                featurePoint.properties = ["highlighted": false, "tailPosition": tailPosition.rawValue, "text": intersection.intersectingWayName, "imageName": imageName, "sortOrder": -index]
+
+                if let intersectingWayName = intersection.intersectingWayName {
+                    annotationCache?.setValue(feature: featurePoint, coordinate: coordinate, intersection: nil, for: intersectingWayName)
+                }
+            }
+            features.append(featurePoint)
+        }
+
+        updateAnnotationLayer(with: FeatureCollection(features: features))
+    }
+
+    private func addAnnotationSymbolImages() {
+        guard let style = mapView.style, style.getStyleImage(with: "AnnotationLeftHanded") == nil, style.getStyleImage(with: "AnnotationRightHanded") == nil else { return }
+
+        // Centered pin
+        if let image = UIImage(named: "AnnotationCentered", in: .mapboxNavigation, compatibleWith: nil) {
+            let stretchX = [ImageStretches(first: Float(20), second: Float(30)), ImageStretches(first: Float(90), second: Float(100))]
+            let stretchY = [ImageStretches(first: Float(26), second: Float(32))]
+            let imageContent = ImageContent(left: 20, top: 26, right: 100, bottom: 33)
```

Comment:
Now that mapbox/mapbox-maps-ios#1269 has landed, we should define these stretch regions in the asset catalog instead of in code, where they’re more fragile.